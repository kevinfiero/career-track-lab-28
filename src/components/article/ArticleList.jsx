import React from 'react';
import PropTypes from 'prop-types';
import { v4 as uuidv4 } from 'uuid';
import Article from './Article';

const ArticleList = ({ articles }) => {
  const articleElements = articles.map(article => (
    <li key={uuidv4()}>
      <Article article={article} />
    </li>
  ));

  return (
    <>
      <ul data-testid="articleList">
        <div>{articleElements}</div>
      </ul>
    </>
  );
};

ArticleList.propTypes = {
  articles: PropTypes.array.isRequired
};

export default ArticleList;
