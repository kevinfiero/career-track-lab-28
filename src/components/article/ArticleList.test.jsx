import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArticleList from './ArticleList';

describe('ArticleList component', () => {
  afterEach(() => cleanup());
  it('renders ArticleList', () => {
    const { asFragment } = render(<ArticleList  
      //bgColor="#FF0000" 
      //fgColor="#0000FF" 
      //text="Hello!" 
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
