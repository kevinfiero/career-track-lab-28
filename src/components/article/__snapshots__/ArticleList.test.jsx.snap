// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component renders ArticleList 1`] = `
<DocumentFragment>
  <ul
    data-testid="articleList"
  >
    <div>
      <li>
        <h1>
          Article 1
        </h1>
      </li>
      <li>
        <h1>
          Article 2
        </h1>
      </li>
      <li>
        <h1>
          Article 3
        </h1>
      </li>
    </div>
  </ul>
</DocumentFragment>
`;
